File: to-do-app/to-do-node/src/controller/routes/routesTasks.ts
import express from 'express';
import taskListController from '../taskListController';
import mid from '../middleware/middleware'

const router = express.Router();

router.get('/api/taskList', mid.validateAuth, taskListController.getTasksAndLists)
router.get('/api/list', mid.validateAuth, taskListController.getLists)
router.get('/api/taskList/tasks/:id', mid.validateAuth, taskListController.getTasksByList)
router.post('/api/taskList', mid.validateAuth, taskListController.createTaskList)
router.put('/api/taskList/:id', mid.validateAuth, taskListController.updateTaskList)
router.delete('/api/taskList/:id', mid.validateAuth, taskListController.deleteTaskList)
router.put('/api/taskList/tasks/complete/:id', mid.validateAuth, taskListController.completeTasksFromList)
router.put('/api/taskList/tasks/uncomplete/:id', mid.validateAuth, taskListController.uncompleteTasksFromList)
router.put('/api/taskList/tasks/moveFromList', mid.validateAuth, taskListController.moveTasksFromList)
router.post('/api/task', mid.validateAuth, taskListController.createTask)
router.put('/api/task/:id', mid.validateAuth, taskListController.updateTask)
router.delete('/api/task/:id', mid.validateAuth, taskListController.deleteTask)
router.put('/api/task/complete/:id', mid.validateAuth, taskListController.completeTask)
router.put('/api/task/uncomplete/:id', mid.validateAuth, taskListController.uncompleteTask)
router.put('/api/task/list/:id', mid.validateAuth, taskListController.moveTaskToList)

export default { router };

File: to-do-app/to-do-go/src/router/routes/routesTasks.go
package routes

import (
	"net/http"

	"renanlelis.github.io/portfolio/to-do-go/src/controller"
)

var tasksRoutes = []Route{
	{
		URI:       "/api/taskList",
		Method:    http.MethodGet,
		Function:  controller.GetTasksAndLists,
		NeedsAuth: true,
	},
	{
		URI:       "/api/list",
		Method:    http.MethodGet,
		Function:  controller.GetLists,
		NeedsAuth: true,
	},
	{
		URI:       "/api/taskList/tasks/{id}",
		Method:    http.MethodGet,
		Function:  controller.GetTasksByList,
		NeedsAuth: true,
	},
	{
		URI:       "/api/taskList",
		Method:    http.MethodPost,
		Function:  controller.CreateTaskList,
		NeedsAuth: true,
	},
	{
		URI:       "/api/taskList/{id}",
		Method:    http.MethodPut,
		Function:  controller.UpdateTaskList,
		NeedsAuth: true,
	},
	{
		URI:       "/api/taskList/{id}",
		Method:    http.MethodDelete,
		Function:  controller.DeleteTaskList,
		NeedsAuth: true,
	},
	{
		URI:       "/api/taskList/tasks/complete/{id}",
		Method:    http.MethodPut,
		Function:  controller.CompleteTasksFromList,
		NeedsAuth: true,
	},
	{
		URI:       "/api/taskList/tasks/uncomplete/{id}",
		Method:    http.MethodPut,
		Function:  controller.UncompleteTasksFromList,
		NeedsAuth: true,
	},
	{
		URI:       "/api/taskList/tasks/moveFromList",
		Method:    http.MethodPut,
		Function:  controller.MoveTasksFromList,
		NeedsAuth: true,
	},
	{
		URI:       "/api/task",
		Method:    http.MethodPost,
		Function:  controller.CreateTask,
		NeedsAuth: true,
	},
	{
		URI:       "/api/task/{id}",
		Method:    http.MethodPut,
		Function:  controller.UpdateTask,
		NeedsAuth: true,
	},
	{
		URI:       "/api/task/{id}",
		Method:    http.MethodDelete,
		Function:  controller.DeleteTask,
		NeedsAuth: true,
	},
	{
		URI:       "/api/task/complete/{id}",
		Method:    http.MethodPut,
		Function:  controller.CompleteTask,
		NeedsAuth: true,
	},
	{
		URI:       "/api/task/uncomplete/{id}",
		Method:    http.MethodPut,
		Function:  controller.UncompleteTask,
		NeedsAuth: true,
	},
	{
		URI:       "/api/task/list/{id}",
		Method:    http.MethodPut,
		Function:  controller.MoveTaskToList,
		NeedsAuth: true,
	},
}
